src/publishProject/dockerUtils.ts

```diff
@@ -0,0 +1,232 @@
+): string[] {
+    const versionToImageTags: Map<number, string[]> = new Map<number, string[]>();
+    let imageTags: string[] | undefined = [];
+    if (rawTags) {
+        // Create a map for version and tags and find the max version in the list
+        let defaultVersion = 0;
+        let maxVersionNumber: number = defaultVersion;
+        (rawTags as string[]).forEach((imageTag) => {
+            const version = findSqlVersionInImageName(imageTag) || defaultVersion;
+            let tags = versionToImageTags.has(version) ? versionToImageTags.get(version) : [];
+            tags = tags ?? [];
+            tags = tags?.concat(imageTag);
+            versionToImageTags.set(version, tags);
+            maxVersionNumber = version && version > maxVersionNumber ? version : maxVersionNumber;
+        });
+
+        // Find the version maps to the target framework and default to max version in the tags
+        const targetVersion =
+            findSqlVersionInTargetPlatform(getTargetPlatformFromVersion(target)) ||
+            maxVersionNumber;
+
+        // Get the image tags with no version of the one that matches project platform
+        versionToImageTags.forEach((tags: string[], version: number) => {
+            if (version === defaultVersion || version >= targetVersion) {
+                imageTags = imageTags?.concat(tags);
+            }
+        });
+
+        imageTags = imageTags ?? [];
+        imageTags = imageTags.sort((a, b) =>
+            a.indexOf(constants.dockerImageDefaultTag) > 0 ? -1 : a.localeCompare(b),
+        );
+
+        if (defaultTagFirst) {
+            const defaultIndex = imageTags.findIndex((i) => i === imageInfo.defaultTag);
+            if (defaultIndex > -1) {
+                imageTags.splice(defaultIndex, 1);
+                imageTags.unshift(imageInfo.defaultTag);
+            }
+        }
+    }
+    return imageTags;
+}
+
+export function getDockerBaseImage(target: string, azureTargetVersion?: string): DockerImageInfo {
+    return {
+        name: `${constants.sqlServerDockerRegistry}/${constants.sqlServerDockerRepository}`,
+        displayName:
+            azureTargetVersion && target === azureTargetVersion
+                ? constants.AzureSqlDbFullDockerImageName
+                : constants.SqlServerDockerImageName,
+        agreementInfo: {
+            link: {
+                text: "Microsoft SQL Server License Agreement",
+                url: constants.sqlServerEulaLink,
+            },
+        },
+        tagsUrl: `https://${constants.sqlServerDockerRegistry}/v2/${constants.sqlServerDockerRepository}/tags/list`,
+        defaultTag: constants.dockerImageDefaultTag,
+    };
+}
+
+/**
+ * Loads Docker tags for a given target version and updates form component options
+ * @param targetVersion project target version
+ * @param tagComponent form component to update with tags
+ * @param formState current form state to set default tag if needed
+ */
+export async function loadDockerTags(
```

Comment:
I'm wary of this entire file.  Why are we using a different method for populating the list of available docker images from the docker deployment UI?  Seems odd that Publish Project fetches the list of available tags from the server, but the deployment UI has it hard-coded.

Virtually all of this logic should be shared between the two features, IMO.

We should be consistent so that 1) users know what to expect, and 2) we can reuse code.  If this is a PM decision, can you have your PM reach out?

Reply:
I'll check what the current deployment UI has.

Initially I moved the method from sql-database-projects and refactored it to get the docker images of selected target version.